### PR TITLE
Menu component and app infrasturcure

### PR DESCRIPTION
**Pull request by yairtawil**
Create the main app infrastructure (angular cli) ....
Add plug-able modules infrastructure to be taken care by the menu container .

### CODE REVIEW

**Comment by yairtawil** [comment]
Oouups..... missing some tests

**Comment by yairtawil** [comment]
That's nice but where the tests you promised :tophat:

File: README.md [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # ansyn
 This is the main repo for all AnSyn modules and apps
 
-#tryies
+#run tests by
```
Comment (yairtawil) [inline README.md]: nice .. 
good job

**Comment by yairtawil** [comment]
Menu component should include "width" input.

File: src/app/app-data/app-data.service.spec.ts [hunk 2]
```diff
@@ -1,11 +1,12 @@
 import { TestBed, inject } from '@angular/core/testing';
 import {AppDataService} from "./app-data.service";
+import {MenuService} from "../menu/menu.service";
```
Comment (osadan) [inline src/app/app-data/app-data.service.spec.ts]: menu service should be the first in line

File: src/app/app-data/app-data.service.spec.ts [hunk 3]
```diff
@@ -1,11 +1,12 @@
...
 
 
 describe('AppDataService', () => {
   beforeEach(() => {
     TestBed.configureTestingModule({
-      providers: [AppDataService]
+      providers: [AppDataService, MenuService]
```
Comment (osadan) [inline src/app/app-data/app-data.service.spec.ts]: this is very good job

**Comment by yairtawil** [comment]
Menu component should include "color" input.

**Pull request merged**